extensions/kubernetes-client/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/graal/CertUtilsSubstitutions.java

```diff
@@ -0,0 +1,18 @@
+package io.quarkus.kubernetes.client.runtime.graal;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.PrivateKey;
+
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+
+@TargetClass(className = "io.fabric8.kubernetes.client.internal.CertUtils")
+public final class CertUtilsSubstitutions {
+
+    // take out this method so we can run the native image without '--allow-incomplete-classpath'
+    @Substitute
+    static PrivateKey handleECKey(InputStream keyInputStream) throws IOException {
+        throw new RuntimeException("EC Keys are not supported when using the native binary");
```

Comment:
What are the consequences of this limitation?

Reply:
BouncyCastle is an optional dependency of the Kubernetes client that we don't bring in and in most cases will never be needed. So I don't anticipate this to cause people any problems in practice.
We could however bring in BouncyCastle as well if we see that this is causing issues (which like I said I doubt will happen). Does that sound acceptable?